### PR TITLE
Migrate to composite actions

### PR DESCRIPTION
**Pull request by ohbarye**
## Issue

closes https://github.com/reviewdog/action-brakeman/issues/15

## Change

As like https://github.com/reviewdog/action-rubocop/pull/29, this pull request tries to migrate to composite actions.

I've tested this change with the following YAML on my repo and that works.

```yaml
name: reviewdog
on: [pull_request]
jobs:
  brakeman:
    name: runner / brakeman
    runs-on: ubuntu-latest
    steps:
      - name: Check out code
        uses: actions/checkout@v1
      - name: Set up Ruby
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: 3.0.3
      - name: brakeman
        uses: ohbarye/action-brakeman@migrate-to-composite-actions
        with:
          brakeman_version: 5.0.2
          github_token: ${{ secrets.github_token }}
          reporter: github-pr-review # Default is github-pr-check
```

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +82,10 @@ jobs:
     steps:
       - name: Check out code
         uses: actions/checkout@v1
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.0.3
```
Comment (ohbarye) [inline README.md:88]: It's kind of a breaking change since this change compels users to define this section.

File: script.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/sh -e
```
Comment (ohbarye) [inline script.sh:1]: fails fast 🔔

File: script.sh [hunk 3]
```diff
@@ -9,16 +9,24 @@ version() {
...
+TEMP_PATH="$(mktemp -d)"
+PATH="${TEMP_PATH}:$PATH"
+
+echo '::group::🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog'
+curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b "${TEMP_PATH}" "${REVIEWDOG_VERSION}" 2>&1
+echo '::endgroup::'
+
+echo '::group:: Installing brakeman with extensions ... https://github.com/presidentbeef/brakeman'
 # if 'gemfile' brakeman version selected
-if [[ $INPUT_BRAKEMAN_VERSION = "gemfile" ]]; then
```
Comment (ohbarye) [inline script.sh:13]: `[[` is one of `bash` expressions. It doesn't work on `sh`.

File: script.sh [hunk 4]
```diff
@@ -1,24 +1,26 @@
-#!/bin/sh
-
-version() {
-  if [ -n "$1" ]; then
-    echo "-v $1"
-  fi
-}
+#!/bin/sh -e
 
 cd "$GITHUB_WORKSPACE"
```
Comment (mgrachev) [inline script.sh:3]:
```suggestion
cd "${GITHUB_WORKSPACE}/${INPUT_WORKDIR}" || exit
```

**Comment by mgrachev** [comment]
@ohbarye Hi 👋

Thank you for your contribution!
Could you add a test to make sure that everything works?
For example: https://github.com/reviewdog/action-rubocop/blob/master/.github/workflows/ci.yml

File: script.sh (hunk 4 above)
Comment (ohbarye) [reply script.sh]: Fixed in 27de346 by adding `workdir` option.

**Review by mgrachev — APPROVED** [review]

**Comment by mgrachev** [comment]
@ohbarye Thanks a lot for your help 🔥

**Pull request merged**